src/libs/actions/Report.js

```diff
@@ -268,7 +269,11 @@ function addActions(reportID, text = '', file) {
         {
             onyxMethod: CONST.ONYX.METHOD.MERGE,
             key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
-            value: _.mapObject(optimisticReportActions, () => null),
+            value: _.mapObject(optimisticReportActions, (action) => {
+                // eslint-disable-next-line no-param-reassign
+                action.errors = {[DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage')};
+                return action;
+            }),
```

Comment:
This is mutating optimisticReportActions's values, which is already in Onyx and possibly read by components.

You should probably instead do to avoid weird bugs

```suggestion
            value: _.mapObject(optimisticReportActions, (action) => {
                return {
                    ...action,
                    errors: {
                        [DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage')
                    }
               };
            }),
```

Reply:
You are right. I thought mapObject already made a copy

Reply:
Oh, but no, you are not right 😄 
This data is not in Onyx yet, it's an object created above in this same method. I think this is fine.

Reply:
> I thought mapObject already made a copy

It should return a new object, but I don't think it makes copies of each value

> This data is not in Onyx yet, it's an object created above in this same method. I think this is fine.
I think it goes into onyx when we merge the Optimistic data 

Ah, I think your are right, the problem should appear earlier on:

- `optimisticReportActions` is referencing some object
- `optimisticReportActions` will go to Onyx as optimistic data
- You create the `failureData` based on `optimisticReportActions`, but mutating it

I think the `errors` key you add will be added to the optimistic data as well because the values in `optimisticReportActions` were not copied, but modified.

Reply:
if there is a problem, I think it should be visible if you just `console.log(optimisticData)` after you initialize `failureData`